analyses/long-format-table-utils/annotator/download-annotation-data.R

```diff
@@ -0,0 +1,205 @@
+# collapse_rp_lists(list(c("NP_1", "NP_2"), c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NX_1", "NA"), c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NP_1", "NP_2"), NULL, c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NP_3", "NP_2")))
+# collapse_rp_lists(list(c("NP_1", "NP_2"), NULL, c("NP_1", "NP_3", NA),
+#                        c(NA, NA, character(0)), c(NA, character(0)),
+#                        character(0)))
+
+
+
+# Collapse a name character vector from mygene.info query results
+#
+# Args:
+# - gn_vec: character vector of name values from mygene.info query results
+#
+# Returns a single character value of a comma-separated name value or NA
+collapse_name_vec <- function(gn_vec) {
+  # assert non-null list elements are all characters
+  if (!is.character(gn_vec)) {
+    stop(paste0("mygene query returns non-character refseq.protein.\n",
+                "Check query results. Revise download-annotation-data.R ",
+                "to handle non-character values."))
+  }
+  # remove NAs
+  uniq_rm_na_gn_vec <- sort(unique(purrr::discard(gn_vec, is.na)))
+
+  clp_gn_str <- paste(uniq_rm_na_gn_vec, collapse = ",")
+  if (identical(clp_gn_str, "")) {
+    # no non-NA name for the gene query
+    # dplyr::summarise needs function return value to be character for a
+    # character column. as.character(NA) is still NA rather than "NA".
+    return(as.character(NA))
+  } else {
+    return(clp_gn_str)
+  }
+}
+# # test cases
+# collapse_name_vec(c("a", "b"))
+# collapse_name_vec(c("a", "b", NA))
+# collapse_name_vec(c("a", "a", "a", "b", "d"))
+# collapse_name_vec(c("d", "a", "a", "b", "d"))
+# collapse_name_vec(c(NA, character(0)))
+# is.na(collapse_name_vec(c(NA, character(0))))
+# class(collapse_name_vec(c(NA, character(0))))
+# collapse_name_vec(c(NA, NA, NA, character(0)))
+# collapse_name_vec(character(0))
+# # folloing cases should fail
+# collapse_name_vec(c(1))
+
+
+
+
+# Set up directory paths -------------------------------------------------------
+# Adapted from the oncoprint-landscape module.
+#
+# Detect the ".git" folder -- this will in the project root directory. Use this
+# as the root directory to ensure proper execution, no matter where it is called
+# from.
+#
+# This only works if the working directory is OpenPedCan-analysis or a
+# subdirectory of OpenPedCan-analysis
+#
+# root_dir is the absolute path of OpenPedCan-analysis
+tryCatch(
+  {
+    root_dir <- rprojroot::find_root(rprojroot::has_file(".git/index"))
+  },
+  error = function(err_cond) {
+    # adapted from http://adv-r.had.co.nz/Exceptions-Debugging.html
+    err_cond$message <- paste0(
+      err_cond$message,
+      "\nTry re-running this script with working directory as ",
+      "OpenPedCan-analysis or a subdirectory of OpenPedCan-analysis.\n")
+    stop(err_cond)
+  }
+)
+
+input_data_dir <- file.path(root_dir, "data")
+
+output_data_dir <- file.path(
+  root_dir, "analyses", "long-format-table-utils", "annotator",
+  "annotation-data")
```

Comment:
I guess I can go back and forth on this but we could make `annotation-data` be the `results` folder instead - thoughts? Although, the onco-kb file is more of an `input` file at this stage since it is manually retrieved.

Reply:
I think `annotation-data` might be easier to understand by the users, as it signifies that the folder contains annotation data rather than results, which is also used in `run-download-annotation-data.sh` and `download-annotation-data.R`, even though the `annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv` is sort of like the results of `download-annotation-data.R`. The `results` directory name may also confuse the users, as they may thing they need to output results there.

We could definitely rename `annotator/annotation-data` to `annotator/results` if preferred.

Reply:
let's keep as `annotation-data`